neurons/validator.py

```diff
@@ -203,6 +210,106 @@ async def competition_loop_tick(self):
         ]
         self.save_state()
 
+    async def fetch_organization_data_references(self):
```

Comment:
I think it's time to have separate file / class for all huggingface operations

Reply:
move all this logic separately, let's keep validator's code to minimum  for readability

Reply:
> I think it's time to have separate file / class for all huggingface operations

Im still sceptic about having all hugging face things somehow aggregated. Its just an api like any other api. The purpose of the function and what (so where also) it serves is what I would favor when considering location.